### ISSUE TITLE
Installation issue: <mumps>

### ISSUE THREAD

**Issue by babouFomb** [LLNL/spack#45128]
### Steps to reproduce the issue

```console
$ spack compiler find
$ spack install mumps%gcc@11.4.00
```

### Error message

<details><summary>Error message</summary>

<pre>
==> Error: FetchError: All fetchers failed for spack-stage-mumps-5.6.2-njceydnqdoorf5buqccnmpt3kxqiinlm
</pre></details>

### Information on your system

```
* **Spack:** 0.22.0 (5fe93fee1eec46a0750bd340198bffcb92ff9eec)
* **Python:** 3.10.12
* **Platform:** linux-ubuntu22.04-skylake
* **Concretizer:** clingo
```

### Additional information

@jcortial-safran 

I'm unable to install mumps since last friday. I get always the FetchError message.
Trying to goon mumps website https://graal.ens-lyon.fr/MUMPS/index.php fails also with the ```PR_CONNECT_RESET_ERROR``` message. 

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

**Comment by jcortial-safran** [comment]
It should be fixed in the develop branch : https://github.com/spack/spack/pull/44897

**Comment by babouFomb** [comment]
Thank. I tried and it the fix proposed in the develop branch : #44897 works.
